content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -184,6 +182,14 @@ for `kube-apiserver`, `kube-controller-manager`, `kube-scheduler` and `etcd` to
 If the flag is not set, those components run as root. You can change the value of this feature gate before
 you upgrade to a newer version of Kubernetes.
 
+`WaitForAllControlPlaneComponents`
+: With this feature gate enabled kubeadm will wait for all control plane components (kube-apiserver,
+kube-controller-manager, kube-scheduler) on a control plane node to report status 200 on their `/healthz`
+endpoints. These checks are performed on `https://127.0.0.1:PORT/healthz`, where `PORT` is taken from
+`--secure-port` of a component. If you specify custom `--secure-port` values in the kubeadm configuration
+they will be respected. Without the feature gate enabled, kubeadm will only wait for the kube-apiserver
+on a control plane node to become ready.
```

Comment:
adding the sentence
> You are advised to enable this feature gate in case you wish to observe a ready
state from all control plane components during the `kubeadm init` or `kubeadm join` command execution.

Reply:
> This text doesn't give the reader a clue about whether we advise them to run with it enabled or disabled (which is different from what the default setting is). Readers also don't get to learn under what circumstances they should think about turning it back off.

i think this is a detail missing in most documented core k8s feature gates.